### PR TITLE
Add documentation for crate features

### LINKED ISSUE #544: no feature flag docs?
I can't find any docs on feature flags.  I am familiar with where to look (Cargo.toml) but [others aren't](https://github.com/bikeshedder/deadpool/issues/23)

### PR DESCRIPTION
**Pull request by bikeshedder**
This closes #544

### CODE REVIEW

File: postgres/src/lib.rs [hunk 1]
```diff
@@ -46,6 +46,20 @@
 //! as an argument. The `NoTls` type in this crate can be used when TLS is not required. Otherwise, the
 //! `postgres-openssl` and `postgres-native-tls` crates provide implementations backed by the `openssl` and `native-tls`
 //! crates, respectively.
+//!
+//! # Features
+//!
+//! The following features can be enabled from `Cargo.toml`:
+//!
+//! | Feature | Description | Extra dependencies | Default |
```
Comment (sfackler) [inline postgres/src/lib.rs]: Given that we don't have any default features on the postgres crate, let's remove the last column.

**Comment by sfackler** [comment]
Thanks! LGTM other than the one change.

**Comment by bikeshedder** [comment]
I left the last column in because it was simpler to copy and paste it from the `tokio-postgres` crate. (Leaving out the `runtime` feature)

If leaving out that column an extra sentence is needed: `No features are enabled by default.`

**Comment by sfackler** [comment]
Fair enough.

**Pull request merged**